cli/packages/util/secrets.go

```diff
@@ -307,27 +307,23 @@ func FilterSecretsByTag(plainTextSecrets []models.SingleEnvironmentVariable, tag
 }
 
 func GetAllEnvironmentVariables(params models.GetAllSecretsParameters, projectConfigFilePath string) ([]models.SingleEnvironmentVariable, error) {
-	isConnected := CheckIsConnectedToInternet()
```

Comment:
This leads back to my prior comment, a bit of clarification on why we aren't doing the backup secrets check inside this function would be great!

Reply:
In our current implementation (before this PR), we are writing to a backup file inside this function which means that every time this function gets called, it overwrites the backup secrets. This becomes problematic with the addition of the `ExpandSecrets` function because it recursively uses the `GetAllEnvironmentVariables` function to fetch secrets in various folder depths. Since the secret backups are scoped to workspace->env, the recursive calls to `GetAllEnvironmentVariables` overwrites the correct values.

<img width="1072" alt="image" src="https://github.com/Infisical/infisical/assets/65645666/83635144-7f08-4c5d-92b7-288e968e75df">

Thinking about it now, we could just change the scoping of the secret backups so that they also include folder paths. What do you think?

Reply:
it should be straightforward since the secrets backup mechanism was built specifically for offline mode support. will do this!